src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -789,15 +797,44 @@ private byte[] GetSidBinaryFromWindows(string domain, string user)
             }
         }
 
+        public bool HasActiveSession(string domainName, string userName)
+        {
+            return EnumerateUsers.IsActiveSessionExists(domainName, userName);
+        }
+
+        public string GetSecurityId(string domainName, string userName)
+        {
+            var account = new NTAccount(domainName, userName);
+            var sid = account.Translate(typeof(SecurityIdentifier));
+            return sid != null ? sid.ToString() : null;
+        }
+
+        public void SetAutoLogonPassword(string password)
+        {
+            using (LsaPolicy lsaPolicy = new LsaPolicy(LSA_AccessPolicy.POLICY_CREATE_SECRET))
+            {
+                lsaPolicy.SetSecretData(LsaPolicy.DefaultPassword, password);
+            }
+        }
+
+        public bool IsRunningInElevatedMode()
+        {
+            return new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator);
+        }
+
         // Helper class not to repeat whenever we deal with LSA* api
         internal class LsaPolicy : IDisposable
         {
             public IntPtr Handle { get; set; }
 
-            public LsaPolicy()
+            public LsaPolicy() 
```

Comment:
@TingluoHuang i'll defer to your expertise here. You're probably a better person to review this.